ml-agents/mlagents/trainers/trainer.py

```diff
@@ -170,59 +173,53 @@ def export_model(self) -> None:
         """
         self.policy.export_model()
 
-    def write_summary(self, global_step: int, delta_train_start: float) -> None:
+    def _write_summary(self) -> None:
         """
         Saves training statistics to Tensorboard.
         :param delta_train_start:  Time elapsed since training started.
         :param global_step: The number of steps the simulation has been going for
         """
-        if (
-            global_step % self.trainer_parameters["summary_freq"] == 0
-            and global_step != 0
-        ):
-            is_training = (
-                "Training."
-                if self.is_training and self.get_step <= self.get_max_steps
-                else "Not Training."
-            )
-            step = min(self.get_step, self.get_max_steps)
-            stats_summary = self.stats_reporter.get_stats_summaries(
-                "Environment/Cumulative Reward"
-            )
-            if stats_summary.num > 0:
-                LOGGER.info(
-                    " {}: {}: Step: {}. "
-                    "Time Elapsed: {:0.3f} s "
-                    "Mean "
-                    "Reward: {:0.3f}"
-                    ". Std of Reward: {:0.3f}. {}".format(
-                        self.run_id,
-                        self.brain_name,
-                        step,
-                        delta_train_start,
-                        stats_summary.mean,
-                        stats_summary.std,
-                        is_training,
-                    )
+        is_training = (
+            "Training."
+            if self.is_training and self.get_step <= self.get_max_steps
```

Comment:
Should the `self.get_step <= self.get_max_steps` just be part of `self.is_training`?

Reply:
It's a bit different, since self.is_training is a on/off bool set by the TC. I made another property called `should_still_train` to capture this logic.